sdk/resourcemanager/Azure.ResourceManager.Core/src/Operation/ArmOperation.cs

```diff
@@ -0,0 +1,111 @@
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
+        /// <remarks>
+        /// <see href="https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning">Details on long running operation object.</see>
+        /// </remarks>
+        public virtual Response WaitForCompletion(CancellationToken cancellationToken = default)
+        {
+            return WaitForCompletion(OperationInternals<Response>.DefaultPollingInterval.Seconds, cancellationToken);
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="pollingInterval"> The polling interval in seconds to check for status. </param>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
+        /// <remarks>
+        /// <see href="https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning">Details on long running operation object.</see>
+        /// </remarks>
+        public virtual Response WaitForCompletion(int pollingInterval, CancellationToken cancellationToken = default)
+        {
+            while (true)
+            {
+                UpdateStatus(cancellationToken);
+                if (HasCompleted)
+                {
+                    return new VoidArmResponse(GetRawResponse());
+                }
+
+                Task.Delay(pollingInterval, cancellationToken).Wait(cancellationToken);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Abstract class for long-running or synchronous applications.
+    /// </summary>
+    /// <typeparam name="TOperations"> The <see cref="OperationsBase"/> to return representing the result of the ArmOperation. </typeparam>
+#pragma warning disable SA1402 // File may only contain a single type
+    public abstract class ArmOperation<TOperations> : Operation<TOperations>
+        where TOperations : notnull
+#pragma warning restore SA1402 // File may only contain a single type
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ArmOperation{TOperations}"/> class for mocking.
+        /// </summary>
+        protected ArmOperation()
+        {
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
```

Comment:
IS 'ArmOperation<TOperations>' the type argument for 'TOperations' in the return type?